src/contracts/atlas/GasAccounting.sol

```diff
@@ -65,32 +65,22 @@ abstract contract GasAccounting is SafetyLocks {
         // to be used to cover what they owe.  This will be subtracted later - tx will revert here if there isn't
         // enough.
 
-        EscrowAccountBalance memory solverEscrow = _balanceOf[solverFrom];
+        uint256 bondedBalance = uint256(_balanceOf[solverFrom].balance);
```

Comment:
Shouldn't this be read from `accessData.bonded` rather?
```suggestion
        uint256 bondedBalance = uint256(accessData[solverFrom].bonded);
```

Reply:
good catch!